Fix custom port assignment

robbavey opened the pull request:
Fixes the ability to use custom ports for rabbitmq: the underlying
march hare library disregards the value of the port parameter
if the :hosts option is set to provide multiple addresses. This
commit changes the setup to use the :addresses option, which
enables a port to be included.

Fixes #53

robbavey committed:
Fix custom port assignment

Fixes the ability to use custom ports for rabbitmq: the underlying
march hare library disregards the value of the port parameter
if the :hosts option is set to provide multiple addresses. This
commit changes the setup to use the :addresses option, which
enables a port to be included.

Fixes #53

robbavey committed:
Added clarifying comment

jsvd commented on lib/logstash/plugin_mixins/rabbitmq_connection.rb:
```diff
@@ -134,6 +133,12 @@ def rabbitmq_settings
+      def addresses_from_hosts_and_port(hosts, port)
+        # Expand host to include port, unless port is already present
+        # (Allowing hosts with port was a previously undocumented feature)
+        hosts.map {|host| host.include?(':') ? host : "#{host}:#{port}"}
```
I believe this won't work with ipv6 (eg: `::1`)

robbavey replied:
Current version of march_hare [doesn't work](https://github.com/ruby-amqp/march_hare/issues/134) with IPv6 either...

I do have a commit ready that should work with IPv6 if/when march_hare is fixed

michaelklishin replied:
@robbavey re: the "if" part, March Hare is open source software. Contributions from users are more than welcome.

michaelklishin replied:
The issue is in the [Java client address parser](https://github.com/rabbitmq/rabbitmq-java-client/issues/385), the same one that was [addressed in Bunny](https://github.com/ruby-amqp/bunny/commit/6be10be1f9b44632ddfd9c645a5d67f5c388b261) previously.

michaelklishin replied:
The root cause was addressed in the Java client. Once a new release is published, a new March Hare one will quickly follow. Thanks for reporting this.

robbavey replied:
@michaelklishin Thank you very much for the speedy resolution of this issue - much appreciated.

robbavey committed:
Adds support for IPv6 address in hosts file

Note:
  IPv6 support will be functional once upstream changes to the
rabbitmq java client are available to the march hare gem. Currently,
any attempt to use IPv6 addresses will result in a
NumberFormatException.

colinsurprenant reviewed (approved):
LGTM

robbavey committed:
Add to changelog, bump version
